knowagebirtreportengine/src/main/java/it/eng/spagobi/engines/birt/BirtImageServlet.java

```diff
@@ -78,16 +79,9 @@ public void service(HttpServletRequest request, HttpServletResponse response) {
 			imageFile = new File(completeImageFileName);
 
 			File parent = imageFile.getParentFile();
-			// Prevent directory traversal (path traversal) attacks
-			if (!imageTmpDir.equals(parent)) {
-				logger.error("Trying to access the file [" + imageFile.getAbsolutePath() + "] that is not inside ${java.io.tmpdir}/birt!!!");
-				throw new SecurityException("Trying to access the file [" + imageFile.getAbsolutePath() + "] that is not inside ${java.io.tmpdir}/birt!!!");
-			}
+			String fileName = imageFile.toString();
 
-			if (!imageFile.exists()) {
-				logger.error("File " + imageFile.getPath() + " not found");
-				return;
-			}
+			PathTraversalChecker.get(parent.toString(), fileName);
```

Comment:
parent is NOT a safe dir, imageDirectory or imageTmpDir are safe.
Filename is a complete path, don't see how PathTraversalChecker.get will work

Reply:
We need to talk about this in today's call.